django_webapps/scripts/db_population/parsers/common_parsers.py

```diff
@@ -88,11 +104,110 @@ def parse_telegram_name(tg_str: str) -> str | None:
     if match_result is not None and len(match_result.group(0)) == len(tg_str):
         return f"@{match_result.group(0)}"
 
-    raise ValueError(f"tg: {tg_str} is not valid")
+    return None
+
+
+def find_any_phone_number(number_str: str) -> list[str]:
+    number_str = number_str.replace(" ", "")
+    regex = re.compile(r"[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}")
```

Comment:
hm... I'm not sure about numbers in curly braces, they may make the regex way too strict. Also, I think you meant `{1,3}` in the first pair of braces (country code?)

I'd rather be more lenient here and fail on the parsing stage

Reply:
I tried that but there were tons of failed numbers so I decided to find just any number and try, this one more or less provide a valid number that we can then try to insert in DB

Reply:
Yes I see that `\+` is optional. But is the exact `{3}` correct? I have some doubts there.
Also, is it really necessary to use brackets before question mark when there's only one option? I don't think I have used it in my regular expressions with question marks, just did `\s?` and I think it worked :) Of course, when there are several options, you need brackets

Reply:
Ah, I see, this one in particular generated by ChatGPT, string {3} in "()" brackets  clause which is correct, I think, the other one also seems to be if we take into account that minimum number of digits in the number is 7, regarding brackets, yes, we are not needed, not sure why he decided to add it

Reply:
I have tried different ones but more or less same result, let's stick with the last one

Reply:
I think the second `{3}` could be correct (area code), although in some European countries area codes only have two digits IIRC, but the first `{3}` surely is wrong, isn't it? Country code can have `{1,3}` digits

Reply:
Also, `{4,6}` seems wrong as well because for example in Berlin there are numbers consisting of 8 digits **after** the area code